src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2004,6 +2004,9 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
             $storeDefinitions = [];
             foreach ($resourceStores as $resourceStore) {
                 $storeDsn = $container->resolveEnvPlaceholders($resourceStore, null, $usedEnvs);
+                if (!($usedEnvs || preg_match('#^[a-z]++://#', $resourceStore))) {
```

Comment:
that should be enough IMHO:
```suggestion
                if (!$usedEnvs && !str_contains($resourceStore, '://')) {
```

Reply:
Done, I've also changed it for the session handler config https://github.com/symfony/symfony/compare/34e5112e9afaf2ef4a9fb348a90b8bfeae71c6db..f53ef173a29e4921f5661369391e30ae980a7b3f